app/recover-role/page.tsx

```diff
@@ -0,0 +1,116 @@
+"use client";
+
+import { useState } from "react";
+
+import { AlertCircle, CheckCircle2 } from "lucide-react";
+import { Hex } from "viem";
+import { useAccount, usePublicClient, useWalletClient } from "wagmi";
+import { useWaitForTransactionReceipt } from "wagmi";
+
+import { Alert, AlertDescription, AlertTitle } from "@/components/ui/alert";
+import { Button } from "@/components/ui/button";
+import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@/components/ui/card";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+
+import { abi as HatsAbi } from "@/lib/hatsAbi";
+
+export default function RecoverRolePage() {
+  const publicClient = usePublicClient();
+  const { data: walletClient, isLoading } = useWalletClient();
+  const { address, isConnected } = useAccount();
+  const [peerAddress, setPeerAddress] = useState("");
+  const [result, setResult] = useState<{ success: boolean; message: string } | null>(null);
+
+  const hatsContractAddress = "0x3bc1A0Ad72417f2d411118085256fC53CBdDd137"; // Sepolia testnet address
```

Comment:
Also something for `constants.ts`

Reply:
let's figure this out when we go cross chain, we might need a handler to support multiple chains

Reply:
SGTM